src/cable/selector.cpp

```diff
@@ -0,0 +1,47 @@
+
+#include <cable/base/cable_base_addr.hpp>
+#include <cable/base/cable_base_obj.hpp>
+#include <cable/asio_ioservice_manager.hpp>
+#include <cable/kind.hpp>
+#include <cable/selector.hpp>
+#include <libs0.hpp>
+
+c_card_selector::c_card_selector(unique_ptr<c_cable_base_addr> && my_addr)
+: m_my_addr(std::move(my_addr))
+{
+	_check_abort(m_my_addr.get()); // not-null
+}
+
+c_card_selector::c_card_selector(const c_card_selector & other)
+: m_my_addr( UsePtr(other.m_my_addr).clone() )
+{
+}
+
+c_card_selector & c_card_selector::operator=(const c_card_selector & other) {
+	m_my_addr = other.m_my_addr->clone();
```

Comment:
ok.

@robertoleksy  that is just for optimization, not needed for correctness right?

done